src/library/scala/util/Using.scala

```diff
@@ -0,0 +1,287 @@
+package scala.util
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import scala.util.control.{ControlThrowable, NonFatal}
+
+/** A utility for performing automatic resource management. It can be used to perform an
+  * operation using resources, after which it will release the resources, in reverse order
+  * of their creation. The resource opening, operation, and resource releasing are wrapped
+  * in a `Try`.
+  *
+  * If more than one exception is thrown by the operation and releasing resources,
+  * the exception thrown ''first'' is returned within the `Try`, with the other exceptions
+  * [[java.lang.Throwable.addSuppressed(Throwable) added as suppressed exceptions]]
+  * to the one thrown first. This is the case ''unless'' a later exception is
+  * [[scala.util.control.NonFatal fatal]], and the one preceding it is not. In that case,
+  * the first exception is added as a suppressed exception to the fatal one, and the fatal
+  * one is thrown. If an exception is a
+  * [[scala.util.control.ControlThrowable ControlThrowable]], no exception will be added to
+  * it as a suppressed exception.
+  *
+  * @example
+  * {{{
+  * val lines: Try[List[String]] = Using(resource1) { r1 =>
+  *   r1.lines.toList
+  * }
+  * }}}
```

Comment:
@Ichoran what are your thoughts on removing the `apply` method, so that there aren't two ways to use it - you have to use a `for` comprehension (or manual `map`/`flatMap`)?

Reply:
This has the added benefit of allowing us to pass the implicit `Resource` on construction

Reply:
I don't feel strongly about it either; I'm just concerned that some others might.

I'm going to leave it as is for now